src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataSetStep.test.tsx

```diff
@@ -0,0 +1,582 @@
+
+    const filters1 = within(subject1Hint.getByTestId('filters')).getAllByRole(
+      'listitem',
+    );
+    expect(filters1).toHaveLength(1);
+    expect(filters1[0]).toHaveTextContent('School type');
+
+    const subject2Hint = within(getDescribedBy(radios[1]));
+    expect(
+      within(subject2Hint.getByTestId('Content')).getByText('Test content 2', {
+        selector: 'p',
+      }),
+    ).toBeInTheDocument();
+    expect(subject2Hint.getByTestId('Geographic levels')).toHaveTextContent(
+      'Local Authority; National',
+    );
+    expect(subject2Hint.getByTestId('Time period')).toHaveTextContent(
+      '2015 to 2020',
+    );
+
+    const indicators2 = within(
+      subject2Hint.getByTestId('indicators'),
+    ).getAllByRole('listitem');
+    expect(indicators2).toHaveLength(1);
+    expect(indicators2[0]).toHaveTextContent('Authorised absence rate');
+
+    const filters2 = within(subject2Hint.getByTestId('filters')).getAllByRole(
+      'listitem',
+    );
+    expect(filters2).toHaveLength(2);
+    expect(filters2[0]).toHaveTextContent('Ethnicity');
+    expect(filters2[1]).toHaveTextContent('FSM');
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Next step',
+      }),
+    ).toBeInTheDocument();
+  });
+
+  test('in simple view, clicking `Next step` calls `onSubmit` with correct values', async () => {
+    const handleSubmit = jest.fn();
+
+    render(
+      <DataSetStep
+        {...wizardProps}
+        subjects={testSubjects}
+        onSubmit={handleSubmit}
+      />,
+    );
+
+    userEvent.click(screen.getByLabelText('Subject 1'));
+
+    expect(handleSubmit).not.toHaveBeenCalled();
+
+    userEvent.click(screen.getByRole('button', { name: 'Next step' }));
+
+    const expected: DatSetFormValues = {
+      subjectId: 'subject-1',
+    };
+
+    await waitFor(() => {
+      expect(handleSubmit).toHaveBeenCalledWith(expected);
+    });
+  });
+
+  test('renders the default view if has `renderHighlightLink`', () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    const radios = within(
+      screen.getByRole('group', {
+        name: 'View all featured tables or select a data set',
+      }),
+    ).getAllByRole('radio');
+    expect(radios).toHaveLength(3);
+    expect(radios[0]).toEqual(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+    expect(radios[1]).toEqual(screen.getByRole('radio', { name: 'Subject 1' }));
+    expect(radios[2]).toEqual(screen.getByRole('radio', { name: 'Subject 2' }));
+
+    expect(
+      screen.getByText(
+        'Please select a data set, you will then be able to see a summary of the data, create your own tables, view featured tables, or download the entire data file.',
+      ),
+    ).toBeInTheDocument();
+  });
```

Comment:
Would suggest we do some other basic checks on things _not_ being rendered e.g. the submit button and the 'More details' for each radio.